coralproject/talk
# [CORL-2674]: Fix React 18 upgrade bug and upgrade again

## What does this PR do?

It fixes the bug related to the React 18 upgrade that was affecting the `MatchMedia` component. The bug is related to changes to effect cleanup timing: https://reactjs.org/blog/2020/08/10/react-v17-rc.html#potential-issues. You can see the change in `onShadowRootWidthChange.ts`, line 73. We needed to capture `shadowRootFirstChild` so that it will not be set to `null` when the cleanup effect runs.

These changes also re-upgrade to React 18 since this bug is addressed.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

To confirm that the bug is fixed, you can use Chrome dev tools to emulate a mobile phone, or you can go to `KeyboardShortcuts` and add `browserInfo.macOS` to line 1363 to get the keyboard shortcuts toolbar to show up on desktop, or you can run Coral locally on your mobile device. Whichever way you prefer to get the keyboard shortcuts toolbar (which uses `MatchMedia`) to show up. Make sure you are on the `All comments` tab on the stream. Click on `My Profile` and see it works as expected with no errors. Go back to the `All comments` tab. Change the sort order and see it works as expected with no errors. Go back to the `All comments` tab. Have at least one featured comment. See that clicking on the `Featured` tab works as expected with no errors.

You can also generally click around the stream/admin/install to test the React 18 upgrade more. 
 
## How do we deploy this PR?
